CHANGELOG.md

```diff
@@ -130,6 +131,13 @@ If we missed any change or there's something you'd like to discuss about this ve
 	- `user_follow_users`
 	- `user_unfollow_users`
 	- `audios_features`
+  + The endpoints which changes parameter from `String` to `&Id<Type>`:
+        - `get_a_show`
+        - `get_an_episode`
+        - `get_shows_episodes`
+  + The endpoint which changes parameter from `Vec<Map<String, Value>>` to `Vec<TrackPositions>`:
+        - `playlist_remove_specific_occurrences_of_tracks`
+- The `Offset` type is now an enum to match API logic, `Offset::Position` is `u32` now (it's not a position in time, it's a position in a playlist, and you can't have both `position` and `uri` fields at the same time).
```

Comment:
If you try to post something like this (using test console: https://developer.spotify.com/console/put-play/):

```json
{
  "context_uri": "spotify:album:5ht7ItJgpBH7W6vJ5BqpPr",
  "offset": {
    "position": 5,
    "uri": "spotify:track:4HbKgKkVeRvpL6yKlRf94E"
  },
  "position_ms": 0
}
```

You get an error:
```json
{
  "error": {
    "status": 400,
    "message": "Illegal offset: should be position XOR uri"
  }
}
```

So this `Offset` object is really an enum. Also, the `position` field is not a time in a song (there's a separate `position_ms` field in the request body for it), it's a position in a play context (e.g. a playlist or an album), so it's really a number, not some duration.

Reply:
Good catch!